multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/SyncFlowableStep.java

```diff
@@ -208,6 +211,31 @@ protected ProcessStepHelper getStepHelper() {
         return stepHelper;
     }
 
+    protected void processServiceCreationFailure(ProcessContext context, CloudServiceInstanceExtended serviceInstance,
+                                               CloudOperationException e) {
+        processServiceActionFailure(context, serviceInstance, e, MessageFormat.format(Messages.COULD_NOT_CREATE_OPTIONAL_SERVICE, serviceInstance.getName()));
+    }
+
+    protected void processServiceUpdateFailure(ProcessContext context, CloudServiceInstanceExtended serviceInstance,
+                                               CloudOperationException e, String errorStep) {
+        processServiceActionFailure(context, serviceInstance, e, MessageFormat.format(Messages.COULD_NOT_UPDATE_OPTIONAL_SERVICE, serviceInstance.getName(), errorStep));
```

Comment:
this method has a lot of parameters, let's try no to use more than 3. 
instead of adding a customErrorMessage can you add this message in the exception.
in the steps you can create a new CloudOperationException with more descriptive message and pass it to this method